kernel/kernel-api/src/main/java/io/delta/kernel/expressions/CollationIdentifier.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.expressions;
+
+import java.util.Optional;
+
+public class CollationIdentifier {
+  public static final String PROVIDER_SPARK = "SPARK";
+  public static final String PROVIDER_ICU = "ICU";
+
+  public static final String ICU_COLLATOR_VERSION = "75.1";
+
+  public static final String DEFAULT_COLLATION_NAME = "UTF8_BINARY";
```

Comment:
docs.

and do these need to be defined on the public API? Basically the public APIs correspond to the the Delta protocol. Do these providers or collation names included in the Delta protocol?